src/client/render/render-core.js

```diff
@@ -148,6 +148,54 @@ let renderCore = function() {
 			}
 		},
 
+		setCameraStyle: function(type) {
+			// If this function is called from an event, use the event origin node to determine what type to become
+			if (this && typeof type !== "string") type = this.dataset.type;
+
+			// Check if we're not already the camera type we try to become
+			if (renderCore.controls && type === renderCore.controls.type) return; // Is already this type.
+
+			// Helper function that copies position and rotation from previously used camera
+			function copyPositionAndRotation(target, source) {
+				target.camera.position.copy(source.camera.position);
+				target.camera.rotation.copy(source.camera.rotation);
+			}
+
+			// Copy over transform data, disable previously used camera / controls, enable new camera / controls
+			switch (type) {
+			case "TrackingCamera":
+				if (renderCore.controls) {
+					copyPositionAndRotation(renderCore.trackingCamera, renderCore.controls);
+					renderCore.controls.disable();
+				}
+				renderCore.controls = renderCore.trackingCamera;
+				break;
+			case "FreeCamera":
+			default:
+				if (renderCore.controls) {
+					copyPositionAndRotation(renderCore.freeCamera, renderCore.controls);
+					renderCore.freeCamera.camera.rotation.z = 0; // Make sure we're not at an angle
+					renderCore.controls.disable();
+				}
+				renderCore.controls = renderCore.freeCamera;
+				break;
+			}
+
+			renderCore.controls.enable();
+
+			if (_viewport) _onCanvasResize();
+
+			// Set selected camera style in DOM
+			domReady.then(() => {
+				let node = document.querySelector(`[data-type=${type}]`);
```

Comment:
Maybe a redundant comment, but if the element's `id` and camera's argument name match (e.g. `freeCamera` for both), then `getElementById` should work here.